### DISCUSSION TITLE
How to save images in epub as files?

### DISCUSSION THREAD

**Discussion by amitbha** [koreader/koreader#11018] [Q&A] [answered]
When long press images in epub, there are only three actions in the menu at the bottom:
- original size
- rotate
- close

How to save out these images as files?

**Comment by jonnyl2** [comment]
You can open the epub in Archive Viewer (long tap in File browser, `Open with...` Archive viewer) and then view and extract image files from there.

**Comment by poire-z** [comment, marked as answer]
#6745 : two-finger large diagonal tap = tap fingers in diagonally opposed corners.
It doesn't save the native image, but the view (zoomed, cropped, or not with blank sides) so you get always an image the size of your screen, mainly so they can be used as screensaver images.

> How to save out these images as files?

That's not really KOReader's main purpose - but with View HTML and the Archive viewer mentionned above, you could extract the native image.

**Reply by amitbha** [reply]
My request of this feature is to facilitate note-taking, cause I want to put the images of interest with the exported highlight note, while KO not support highlight images.

I know the EPUB file is a zip file, but it is troublesome to locate the desired images.

At present, there has been a boom in bi-directional linking note-taking methods, and effective reading is inseparable from taking-notes, so it would be great if KO could pay more attention to the note-taking function, and I noticed that displaying the full content of highlights in bookmark is really a good idea.

**Reply by hius07** [reply]
I need cpp `getImageXPointer()` to think in this direction.

**Reply by poire-z** [reply]
Not interested in thinking much about it :/
But I assume just saving the image is not enough: if it's some text with inline images like maths bit as PNG or SVG, you'd want to save the whole thing with images at the right places among the text.
Also, our exporters would have to support that.
I also don't really want more noise/buttons in ImageViewer :)

**Reply by ryanwwest** [reply]
For devices supporting screenshots, maybe that would be the easiest solution for all this? Then you can choose whether to include caption text in the screenshot as well by editing the screenshot.
